src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        for (const data in extraData) {
+          stateCopy[data] = extraData[data];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete stateCopy[key];
```

Comment:
You should also me it work only if State copy actualy has the property you wanna delete.
"if (stateCopy.hasOwnProperty(key)"
Otherwise, it could potentialy cause some problem. Overall, realy good code bro ;)

Reply:
yeah, that's true, as those are keys pointed by programmer / user of the code.